src/Illuminate/Cache/SessionStore.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace Illuminate\Cache;
+
+use Illuminate\Contracts\Cache\LockProvider;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\InteractsWithTime;
+
+class SessionStore extends TaggableStore implements LockProvider
+{
+    use InteractsWithTime, RetrievesMultipleKeys;
+
+    /**
+     * The session instance.
+     *
+     * @var \Illuminate\Session\Store
+     */
+    public $session;
+
+    /**
+     * Indicates if values are serialized within the store.
+     *
+     * @var bool
+     */
+    protected $serializesValues;
+
+    /**
+     * Create a new Session store.
+     *
+     * @param  \Illuminate\Session\Store  $session
+     * @param  bool  $serializesValues
+     */
+    public function __construct($session, $serializesValues = false)
+    {
+        $this->session = $session;
+        $this->serializesValues = $serializesValues;
+    }
+
+    /**
+     * Get all of the cached values and their expiration times.
+     *
+     * @param  bool  $unserialize
+     * @return array<string, array{value: mixed, expiresAt: float}>
+     */
+    public function all($unserialize = true)
+    {
+        if ($unserialize === false || $this->serializesValues === false) {
+            return $this->session->get('_storage', []);
+        }
+
+        $storage = [];
+
+        foreach ($this->session->get('_storage', []) as $key => $data) {
+            $storage[$key] = [
+                'value' => unserialize($data['value']),
+                'expiresAt' => $data['expiresAt'],
+            ];
+        }
+
+        return $storage;
+    }
+
+    /**
+     * Retrieve an item from the cache by key.
+     *
+     * @param  string  $key
+     * @return mixed
+     */
+    public function get($key)
+    {
+        if ($this->session->missing("_storage.{$key}")) {
```

Comment:
I would prefer using `Session@exists()` over `Session@missing()`, as the `exists()` method is part of the `Illuminate\Contracts\Session\Session` interface, whilst `missing()` isn't.